### DISCUSSION TITLE
Optionally suppress "URL ... isn't matching any of your page routes" logs

### DISCUSSION THREAD

**Discussion by crummy** [vikejs/vike#1012] [Discussion]
Following up from #987, I have my 404s redirecting fine but I see a lot of log messages like this:

```
[vite-plugin-ssr@0.4.131][Info] URL [...] isn't matching any of your page routes. See https://vite-plugin-ssr.com/routing for more information about routing. Set the environment variable DEBUG=vps:routing to inspect your app's routing. (This log isn't shown in production.) Your page routes:
[...]
```

My local dev script looks like this:
```
// This file does two things:
// 1. Serves vite+ssg content
// 2. Redirects to earth on any 404s
// Use it for development so you can have just a single entrypoint. Run it with `node local-dev.js`

// This file isn't processed by Vite, see https://github.com/brillout/vite-plugin-ssr/issues/562
//  - Consequently, the server needs be manually restarted when changing this file

import express from "express";
import { renderPage } from "vite-plugin-ssr/server";
const httpProxy = require("http-proxy");

startServer();
const earthProxy = httpProxy.createProxyServer();

const EARTH = "http://localhost:4568";

async function startServer() {
    const app = express();

    // Vite integration
    // We instantiate Vite's development server and integrate its middleware to our server.
    const vite = await import("vite");
    const viteDevMiddleware = (
        await vite.createServer({
            configFile: "./vite.ssg.config.ts",
            server: { middlewareMode: true },
        })
    ).middlewares;
    app.use(viteDevMiddleware);

    // Vite-plugin-ssr middleware. It should always be our last middleware (because it's a
    // catch-all middleware superseding any middleware placed after it).
    app.all("*", async (req, res, next) => {
        const pageContextInit = {
            urlOriginal: req.originalUrl,
        };
        const pageContext = await renderPage(pageContextInit);
        const { httpResponse } = pageContext;
        if (!httpResponse) return next();
        const { body, statusCode, contentType, earlyHints } = httpResponse;

        // our earth redirect. possibly this should be another middleware
        if (statusCode === 404) {
            earthProxy.web(req, res, { target: EARTH });
            return;
        }

        if (res.writeEarlyHints)
            res.writeEarlyHints({
                link: earlyHints.map((e) => e.earlyHintLink),
            });
        res.status(statusCode).type(contentType).send(body);
    });

    const port = 5173;
    app.listen(5173, "0.0.0.0");
    console.log(`Server running at http://localhost:${port}`);
}
```

Is there some way I can intercept earlier, or suppress these log messages?

**Comment by brillout** [comment]
I'm currenlty working on https://github.com/brillout/vite-plugin-ssr/issues/926.

The logging is also improved.

```js
`throw ${abortCaller}() successfully handled while rendering URL '${urlOriginal}' (this log isn't shown in production)`
```

If you don't mind waiting a bit, then let me finish #926 which should fix the situation. (ETA tomorrow, at the latest this week.)

**Reply by crummy** [reply]
Thanks, I don't mind ignoring the logs for a little while.

**Reply by brillout** [reply]
Actually, are you even using [`RenderErrorPage()`](https://vite-plugin-ssr.com/RenderErrorPage)? Because the changes I'm currenlty working on only affect those who use it.

I just remembered your setup and I'm not sure the changes will actually remove the log for you.

**Reply by crummy** [reply]
Here's a repro of our setup (though we use v1): https://github.com/crummy/vike-404-proxy

We are not using RenderErrorPage(), not sure what that is.

**Reply by brillout** [reply]
In that case, a new feature is needed to suppress the log. I went ahead and created a ticket for it: https://github.com/brillout/vite-plugin-ssr/issues/1013. It isn't a trivial change and there are a lot of higher priority tickets, so I'm not sure when it'll be implemented.

Btw. is [sponsoring](https://github.com/sponsors/brillout) something your employer would be up to? It could make sense since you're using unique vite-plugin-ssr added values (I don't think you could achieve your setup with other tools).

**Reply by crummy** [reply]
I'll talk to my boss but we are very tiny. I'll look into contributing it myself if I can find the time.

Do you think suppressing the log is the right way to do this? Or should I look into proxying earlier?

**Reply by brillout** [reply]
I don't see a way for VPS to detect that you're expecting 404s, that's why I think there isn't another way other than supressing.

**Reply by brillout** [reply]
As for sponsoring, feel free to start with a tiny amount. I'll appreciate the gesture.
